app/src/main/java/com/ncnf/main/MainActivity.java

```diff
@@ -17,37 +20,76 @@
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
         setContentView(R.layout.activity_main);
 
-        if (savedInstanceState == null) {
-            getSupportFragmentManager().beginTransaction()
-                    .replace(R.id.mainFragmentContainerView, new HomeFragment())
-                    .commit();
+        if(savedInstanceState == null){
+            this.homeFragment = new HomeFragment();
+            this.mapFragment = new MapFragment();
+            this.feedFragment = new FeedFragment();
+            this.activeFragment = this.homeFragment;
         }
 
+        fragmentManager.beginTransaction().add(R.id.mainFragmentContainerView, feedFragment).hide(feedFragment).commit();
+        fragmentManager.beginTransaction().add(R.id.mainFragmentContainerView, mapFragment).hide(mapFragment).commit();
+        fragmentManager.beginTransaction().add(R.id.mainFragmentContainerView, homeFragment).hide(homeFragment).commit();
+
+        fragmentManager.beginTransaction().show(activeFragment).commit();
+
         navigationBar = findViewById(R.id.mainNavigationBar);
 
-        navigationBar.setOnNavigationItemSelectedListener(item -> {
-            if(item.getItemId() == R.id.navigation_feed){
-                getSupportFragmentManager().beginTransaction()
-                        .replace(R.id.mainFragmentContainerView, new FeedFragment())
-                        .commit();
+        navigationBar.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener);
+    }
+
 
-            } else if(item.getItemId() == R.id.navigation_map) {
-                getSupportFragmentManager().beginTransaction()
-                        .replace(R.id.mainFragmentContainerView, new MapFragment())
-                        .commit();
+    public final  BottomNavigationView.OnNavigationItemSelectedListener mOnNavigationItemSelectedListener = item -> {
+        switch (item.getItemId()){
+            case R.id.navigation_home:
+                fragmentManager.beginTransaction().hide(activeFragment).show(homeFragment).commit();
+                activeFragment = homeFragment;
+                return true;
 
-            } else if(item.getItemId() == R.id.navigation_home){
-                getSupportFragmentManager().beginTransaction()
-                        .replace(R.id.mainFragmentContainerView, new HomeFragment())
-                        .commit();                }
+            case R.id.navigation_map:
+                fragmentManager.beginTransaction().hide(activeFragment).show(mapFragment).commit();
+                activeFragment = mapFragment;
+                return true;
+
+            case R.id.navigation_feed:
+                fragmentManager.beginTransaction().hide(activeFragment).show(feedFragment).commit();
+                activeFragment = feedFragment;
+                return true;
+        }
+
+        return false;
+    };
```

Comment:
Instead of returning false, add default statement to switch case

Reply:
Done !